testing/upload_validation.ts

```diff
@@ -933,137 +943,141 @@ const genericObjectSchema: z.ZodTypeAny = z.lazy(() =>
+      validateSnippetProperty();
+      (schema.subtitleProperties || []).forEach((f, i) => {
+        if (typeof f === 'string') {
+          if (!(f in schema.properties)) {
             context.addIssue({
               code: z.ZodIssueCode.custom,
-              path: ['linkProperty'],
-              message: `The "linkProperty" field name "${schema.linkProperty}" does not exist in the "properties" object.`,
+              path: ['subtitleProperties', i],
+              message: `The "subtitleProperties" field name "${f}" does not exist in the "properties" object.`,
             });
             return;
           }
 
-          const linkPropertySchema = schema.properties[schema.linkProperty];
-          if (linkPropertySchema.type !== ValueType.String || linkPropertySchema.codaType !== ValueHintType.Url) {
-            context.addIssue({
-              code: z.ZodIssueCode.custom,
-              path: ['linkProperty'],
-              message: `The "linkProperty" field name "${schema.linkProperty}" must refer to a "ValueType.String" property with a "ValueHintType.Url" "codaType".`,
-            });
+          const subtitlePropertySchema = schema.properties[f];
+
+          if (!('codaType' in subtitlePropertySchema && subtitlePropertySchema.codaType)) {
             return;
           }
-        }
-      };
 
-      validateTitleProperty();
-      validateLinkProperty();
-      validateImageProperty();
-      validateDescriptionProperty();
-      (schema.subtitleProperties || []).forEach((f, i) => {
-        if (!(f in schema.properties)) {
-          context.addIssue({
-            code: z.ZodIssueCode.custom,
-            path: ['subtitleProperties', i],
-            message: `The "subtitleProperties" field name "${f}" does not exist in the "properties" object.`,
-          });
-          return;
-        }
-
-        const subtitlePropertySchema = schema.properties[f];
-        if (
-          'codaType' in subtitlePropertySchema &&
-          subtitlePropertySchema.codaType &&
-          [
-            ValueHintType.ImageAttachment,
-            ValueHintType.Attachment,
-            ValueHintType.ImageReference,
-            ValueHintType.Embed,
-            ValueHintType.Scale,
-            ValueHintType.Scale,
-          ].includes(subtitlePropertySchema.codaType)
-        ) {
-          context.addIssue({
-            code: z.ZodIssueCode.custom,
-            path: ['subtitleProperties', i],
-            message: `The "subtitleProperties" field name "${f}" must refer to a value that does not have a codaType corresponding to one of ImageAttachment, Attachment, ImageReference, Embed, Scale, or Scale.`,
-          });
-          return;
+          switch (subtitlePropertySchema.codaType) {
+            case ValueHintType.ImageAttachment:
+            case ValueHintType.Attachment:
+            case ValueHintType.ImageReference:
+            case ValueHintType.Embed:
+            case ValueHintType.Scale:
```

Comment:
yeah i was mostly wondering what progress and slider would look like and then fine-tune this since i haven't gotten a chance to test (and this was the existing behavior) 

toggle is another in-between where it's a bit richer than just text but also feels more useful than leaving it raw as a boolean of true or false.

I'd guess we probably don't want to start with any rich representations of data like this but wanted to show the team first